rfc/rfc022-discovery-service.md

```diff
@@ -0,0 +1,299 @@
+It's a _holder_ credential, meaning it has been self issued. Fields under `credentialSubject` (except `id`) can be used by clients depending on the use case.
+The credential has the following requirements:
+- `type` MUST be `DiscoveryRegistrationCredential`.
+- `@context` MUST contain the Nuts JSON-LD context (https://nuts.nl/credentials/v1).
+- `credentialSubject` MUST contain a JSON object with additional parameters.
+- `credentialSubject.id` MUST be the same as the `issuer` of the credential.
+- `id` MUST be a unique identifier for the credential.
+- `issuanceDate` MUST be a valid date.
+
+## 5. Service Definition
+
+Servers MUST share a JSON document describing the discovery service. This document is known as the _service definition_.
+The document MUST contain the following properties:
+- `id` REQUIRED. String containing a value that identifies the service definition. MAY be used to version the definition.
+- `did_methods` OPTIONAL. Array of strings containing the DID methods enabled for the service. If not present, all DID methods are allowed.
+- `endpoint` REQUIRED. String containing the URL where the server hosts the service endpoint.
+- `presentation_definition` REQUIRED. JSON object with the Presentation Definition (see [PEX]) describing requirements for presentations on the list.
+- `presentation_max_validity` REQUIRED. JSON number containing the maximum validity period (number of seconds between `nbf` and `exp`) of a presentation in seconds.
+
+For example:
+
+``json
+{
+  "id": "uc_university_v1",
+  "endpoint": "https://example.com/usecase/university/v1",
+  "presentation_max_validity": 259200,
+  "presentation_definition": {
+    "id": "pd_university",
+    "input_descriptors": [
+      {
+        "id": "pd_university_type",
+        "constraints": {
+          "fields": [
+            {
+              "path": ["$.type"],
+              "filter": {
+                "type": "string",
+                "const": "UniversityCredential"
+              }
+            },
+            {
+              "path": "$.credentialSubject.name",
+              "filter": {
+                "type": "string"
+              }
+            }
+          ]
+        }
+      }
+    ]
+  }
+}
+``
+
+## 6. Trust
+
+Trust of credential issuers (e.g. `did:example:education-accreditor` issuing `EducationalInstitutionCredential`) SHOULD be defined by the presentation definition.
+In this case, there should be 2 constraints: one for the type and one for the issuer:
+
+``json
+[
+  {
+    "path": ["$.type"],
+    "filter": {
+      "type": "string",
+      "const": "EducationalInstitutionCredential"
+    }
+  },
+  {
+    "path": "$.issuer",
+    "filter": {
+      "type": "string",
+      "const": "did:example:education-accreditor"
+    }
+  }
+]
+``
+
+## 7. Security Considerations
+
+## 7.1 Denial of Service
+
+Since the server is hosting a public endpoint, it should take the following measures to prevent denial-of-service attacks:
```

Comment:
Should this apply to all nodes?
All nodes have the endpoint exposed and forward to what they consider the server. (Implementation detail of the nuts-node, so may need to be added to documentation on the node if valid.)

Reply:
according to the spec, clients don't expose anything.